tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -356,9 +356,6 @@ def test_check_changed_artifact_then_download(wandb_init, tmp_path, monkeypatch)
         assert file1.is_file()
         assert file1.read_text() == "hello"
 
-    # Delete the cached file
-    shutil.rmtree(cache_dir)
```

Comment:
Why did you remove this?

Reply:
During this test being run, the cache file that was created in the second run was also getting deleted before we could write the downloaded data to it. 

I'm actually not sure why we're deleting the cache in the first place..Isn't the test to test that we replace modified artifact data?

Reply:
This sounds like bad test isolation then. Each test should use its own cache directory.